main.py

```diff
@@ -0,0 +1,117 @@
+from datetime import datetime
+"""
+open all 3 files
+parse such that we have the log in a list
+count how many lines are in log, level=error count, level=info
+
+TIME FORMAT : 08:56:35.354
+
+top 5 most requested HTTP paths/endpoints
+"""
+
+class Logging:
+    def __init__(self):
+        self.log = []
+        self.date_time = []
+
+
+    def logs(self, file1, file2, file3):
+        file_1 = open(file1, 'r')
+        file_2 = open(file2, 'r')
+        file_3 = open(file3, 'r')
+
+        for line in file_1:
+            self.log.append(line)
```

Comment:
While this achieves the goal of not having to re-ingest the logs everytime you run a function, it misses a bigger brained step.

If you are gonna ingest the logs, you might as well parse them here at the same time. Parsing the logs into it's own type gives you some ways to quickly answer questions about the data set.

```python
class Log:
    def __init__():
        self.level = ""
        self.msg = ""
        self.timestamp = ""
```

Also I see you keep a separate list for datetime. This model also solves this.